internal/controller/storageclient_controller.go

```diff
@@ -665,6 +710,36 @@ func removeStorageClaimAsOwner(obj client.Object) {
 	}
 }
 
+func (r *storageClientReconcile) reconcileResourceByGVK(
+	gvk schema.GroupVersionKind,
+	obj client.Object,
+	resourceMapping gvkToRawresources,
```

Comment:
You don't need the entire map, just the list of resource bytes your are going to reconcile